app/src/main.ts

```diff
@@ -168,6 +169,52 @@ if (shouldQuit) {
           });
         });
       });
+
+      if (isOSX() && appArgs.accessibilityPrompt) {
+        const mediaKeys = [
+          'MediaPlayPause',
+          'MediaNextTrack',
+          'MediaPreviousTrack',
+          'MediaStop',
+        ];
+        const globalShortcutsKeys = appArgs.globalShortcuts.map((g) => g.key);
+        const mediaKeyWasSet =
+          globalShortcutsKeys.filter((g) => mediaKeys.includes(g)).length > 0;
+        if (
+          mediaKeyWasSet &&
+          !systemPreferences.isTrustedAccessibilityClient(false)
+        ) {
+          // Since we're tryig to set global keyboard shortcuts for media keys, we need to prompt
+          // the user for permission on Mac.
+          // For reference:
+          // https://www.electronjs.org/docs/api/global-shortcut?q=MediaPlayPause#globalshortcutregisteraccelerator-callback
+          //
+          const accessibilityPromptResult = dialog.showMessageBoxSync(null, {
+            type: 'question',
+            message: 'Accessibility Permissions Needed',
+            buttons: ['Yes', 'No', 'Never Ask Me Again'],
+            defaultId: 0,
+            detail:
+              `${appArgs.name} would like to use one or more of your keyboard's media keys (start, stop, next track, or previous track) to control it.\n\n` +
+              `Would you like Mac OS to ask for your permission to do so?\n\n` +
+              `If so, you will need to restart ${appArgs.name} after granting permissions for these keyboard shortcuts to begin working.`,
```

Comment:
I'd rather have a simple `Yes` / `Not now` dialog, without this "Never ask me again" option, and an extra line that says something like:

```
If you don't want to / cannot grant ${appArgs.name} Accessibility permissions, please delete the global shortcuts using media keys to not be prompted again.
```

A macOS user who wants to use a media key in a global shortcut has no choice but to comply to grant accessibility permissions, correct?

So, why this new `no-accessibility-prompt` flag, and the need to persist it?

To me, users confronted to this dialog will fall into two buckets:

1. "Accepters", who want their media key global shortcut and accept to enable it at macOS level
2. "Denyers", who would like their media key global shortcut, but for some reason don't want to enable the accessibility setting, and should then simply remove the shortcut from their shortcut options.

Why make things more complex with a third bucket, a new flag, and more persistence needed? You don't want to enable the accessibility thing, alright, we tell you why you can't use your global shortcut. Want us to stop telling you? A. do the OS requirement, or B. get rid of the binding (which at that point was broken anyway). Simple and doesn't need an extra flag, of which Nativefier already has waaaaaaaaaaaaaay too many 😄.

Am I missing something?

Reply:
I could be argued out of it, I was just trying to accomplish two things:
1) If someone is creating an app for distribution, they could disable this prompt if they felt like it would be too intrusive.
2) If someone was downloading a distributed app, they could opt out of this feature without being pestered every time they launch it.

I do get the argument that Nativefier has too many flags though. 😆

Reply:
Okay, now I understand. Take 2: what about reworking it so that you keep the “do not ask again” persistence, but without the extra flag?

That way, someone using a distributed bundle will at least be warned once that the keyboard shortcut cannot work, and will be able to decide what to do. It’s a pestering, but a single/minor one, and it allows for a bit of feedback / communication between users and packagers. Else, the shortcut will simply be invisibly dead to bundle users.

With a bit of tweaking the dialog box text, I think it can be made friendly and understandable.

Waddayathink?

Reply:
That sounds like a fair compromise. :)

How do you think we should tweak the dialog text in that case? I was trying to communicate to the user that they were about to get a big scary dialog asking for permissions that don't seem particularly related to the app itself, even though they are. However, I get that just because it's clear to me doesn't mean it will be to an end user. 😆

Reply:
Command option removed.